storybookjs/storybook
# Interactions: Fix Subnav positioning and overflow states

This resolves two issues with the `Subnav` component:

1. After wrapping the `Subnav` component with the `Bar` component, the `SubNav` would no longer would stick to the top of the panel. I resolved this by adding `SubnavWrapper` with `position: sticky` and other styles needed.
1. When the panel was smaller in width, the `StyledLocation` text would wrap looking quite broken. I resolved that by adding `white-space: nowrap`. That then exposed a secondary issue where the background color of `Subnav` would not extend fully. `SubnavWrapper` also resolved this issue.

## How to test
1. Open the Official Storybook from the link below
1. Navigate to Interactions > Standard email success
1. Open the Interactions addon panel
1. With the panel docked to the bottom, make the height of the panel rather short and make sure the `Subnav`'s `position: sticky` is working properly.
1. With the panel docked to the side (hotkey: `d`), make the panel very narrow and ensure the Location text does not wrap.

### Before
https://user-images.githubusercontent.com/1123119/138134130-ff380236-89d6-4520-a5ff-5635075a800b.mp4

### After
https://user-images.githubusercontent.com/1123119/138134148-8fd99aa4-0771-4b97-b28a-c711ee7c27d9.mp4

<hr>




┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200510412866230/1201237799725888) by [Unito](https://www.unito.io)
